aopy/analysis.py

```diff
@@ -41,26 +46,111 @@ def factor_analysis_dimensionality_score(data_in, dimensions, nfold, maxiter=100
 
         # Handle the case without cross validation.
         if nfold == 1:
-            fa = FactorAnalysis(n_components=dimensions[dim_idx], max_iter = maxiter)
+            fa = FactorAnalysis(n_components=dimensions[dim_idx], max_iter=maxiter)
             fafit = fa.fit(data_in.T)
-            log_likelihood_score[dim_idx,fold_idx] = fafit.score(data_in.T)
-            iterations_required[dim_idx,fold_idx] = fafit.n_iter_
+            log_likelihood_score[dim_idx, fold_idx] = fafit.score(data_in.T)
+            iterations_required[dim_idx, fold_idx] = fafit.n_iter_
             warnings.warn("Without cross validation the highest dimensional model will always fit best.")
-        
+
         # Every other case with cross validation
         else:
             for trainidx, testidx in model_selection.KFold(n_splits=nfold).split(data_in.T):
-                fa = FactorAnalysis(n_components=dimensions[dim_idx], max_iter = maxiter)
-                fafit = fa.fit(data_in[:,trainidx].T)
-                log_likelihood_score[dim_idx,fold_idx] = fafit.score(data_in[:,testidx].T)
-                iterations_required[dim_idx,fold_idx] = fafit.n_iter_
+                fa = FactorAnalysis(n_components=dimensions[dim_idx], max_iter=maxiter)
+                fafit = fa.fit(data_in[:, trainidx].T)
+                log_likelihood_score[dim_idx, fold_idx] = fafit.score(data_in[:, testidx].T)
+                iterations_required[dim_idx, fold_idx] = fafit.n_iter_
                 fold_idx += 1
-        
+
         if verbose == True:
-            print(str((100*(dim_idx+1))//len(dimensions)) + "% Complete")
-            
+            print(str((100 * (dim_idx + 1)) // len(dimensions)) + "% Complete")
+
     return log_likelihood_score, iterations_required
 
+
+'''
+Curve fitting
+'''
+
+
+# These functions are for curve fitting and getting modulation depth and preferred direction from firing rates
+def func(target, b1, b2, b3):
+    theta = 45 * (target - 1)
+    return b1 * np.cos(np.deg2rad(theta)) + b2 * np.sin(np.deg2rad(theta)) + b3
+
+
+def get_modulation_depth(b1, b2):
+    return np.sqrt((b1 ** 2) + (b2 ** 2))
+
+
+def get_preferred_direction(b1, b2):
+    return np.arctan2(b2 ** 2, b1 ** 2)
+
+
+def get_mean_fr_per_direction(data, target):
+    means_d = []
+    stds_d = []
+
+    for i in range(1, 9):
+        means_d.append(np.mean(data[target == i], axis=(0, 2)))
+        stds_d.append(np.std(data[target == i], axis=(0, 2)))
+
+    return means_d, stds_d
+
+
+def plot_mean_fr_per_target_direction(means_d, neuron_id, ax, color, this_alpha, this_label):
+    sns.set_context('talk')
+    ax.plot(np.array(means_d)[:, neuron_id], c=color, alpha=this_alpha, label=this_label)
+
+    ax.legend()
+    ax.set_xlabel("Target", fontsize=16)
+    ax.set_ylabel("Spike Rate (Hz)", fontsize=16)
+    plt.tight_layout()
+
+
+def run_curvefitting(means, make_plot=True, fig_title='Tuning Curve', n_rows=5, n_cols=11):
```

Comment:
not sure if n_rows=5 and n_cols=11 makes sense as a default, since means could be any shape. this seems more specific to the analysis you were doing at the time than would make sense for aopy

maybe remove the plotting feature and just have this function do the curve fitting?